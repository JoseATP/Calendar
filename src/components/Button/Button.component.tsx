import { ButtonHTMLAttributes, FC } from "react";
import {
  BaseButton,
  GradientButton,
  GrayButton,
  RedButton,
} from "./Button.styles";

export enum BUTTON_TYPE_CLASSES {
  base = "base",
  grayButton = "grayButton",
  redButton = "redButton",
  gradientButton = "gradientButton",
}

const getButton = (buttonType = BUTTON_TYPE_CLASSES.base): typeof BaseButton =>
  ({
    [BUTTON_TYPE_CLASSES.base]: BaseButton,
    [BUTTON_TYPE_CLASSES.grayButton]: GrayButton,
    [BUTTON_TYPE_CLASSES.gradientButton]: GradientButton,
    [BUTTON_TYPE_CLASSES.redButton]: RedButton,
  }[buttonType]);

export type ButtonProps = {
  backgroundColor?: string;
  buttonType?: BUTTON_TYPE_CLASSES;
  disabled?: boolean;
} & ButtonHTMLAttributes<HTMLButtonElement>;

const Button: FC<ButtonProps> = ({
  children,
  buttonType,
  disabled,
  ...otherProps
}) => {
  const CustomButton = getButton(buttonType);
  return (
    <CustomButton disabled={disabled} {...otherProps}>
      {children}
    </CustomButton>
  );
};

export default Button;
