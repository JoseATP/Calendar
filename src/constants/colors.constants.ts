export const colors = {
  base: "#394ABC",
  base2: "#2f3c94",
  blackLight: "#384042",
  grey: "#626262",
  greyLight: "#959595",
  white: "#FFFFFF",
  error: "#D52222",
};
