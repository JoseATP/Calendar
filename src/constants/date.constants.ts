export const DATE_OPTIONS: Intl.DateTimeFormatOptions = {
  weekday: "long",
  year: "numeric",
  month: "long",
  day: "numeric",
};
