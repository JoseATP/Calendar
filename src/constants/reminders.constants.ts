export const remindersData = [
  {
    id: "0",
    title: "Standups and DS&T",
    description:
      "Codelitt Inc. is inviting to a Zoom meeting in which everyone as people from the company will participate.",
    date: new Date(2023, 3, 3).toISOString().split("T")[0],
    hour: "11:15",
    color: "#C8E6C9",
  },
  {
    id: "1",
    title: "Architecture Overview",
    description:
      "Meeting for the presentation of the new application architecture.",
    date: new Date(2023, 3, 3).toISOString().split("T")[0],
    hour: "12:00",
    color: "#FFCC80",
  },
  {
    id: "2",
    title: "Interview",
    description: "Interview for the DevOps position",
    date: new Date(2023, 3, 3).toISOString().split("T")[0],
    hour: "14:40",
    color: "#5BA4CF",
  },
  {
    id: "3",
    title: "Interview",
    description: "Interview for the DevOps position",
    date: new Date(2023, 3, 4).toISOString().split("T")[0],
    hour: "14:40",
    color: "#6DECA9",
  },
];
