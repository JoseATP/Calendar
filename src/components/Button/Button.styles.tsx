import styled from "styled-components";
import { colors } from "../../constants/colors.constants";

type BaseButtonProps = {
  backgroundColor?: string;
};

export const BaseButton = styled.button<BaseButtonProps>`
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 16px;
  background-color: ${colors.base};
  color: ${colors.white};
  font-family: "Open Sans";
  font-weight: bold;
  border-radius: 5px;
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    background: linear-gradient(90deg, ${colors.base2} 0%, ${colors.base} 100%);
  }
`;

export const GrayButton = styled(BaseButton)`
  background-color: ${colors.grey};
  color: ${colors.white};

  &:hover {
    background: linear-gradient(90deg, #333333 0%, #626262 100%);
  }
`;

export const RedButton = styled(BaseButton)`
  background-color: ${colors.error};
  color: ${colors.white};

  &:hover {
    background: linear-gradient(90deg, #da4848 0%, ${colors.error} 100%);
  }
`;

export const GradientButton = styled(BaseButton)`
  background: linear-gradient(90deg, #ff465d 0%, #bc46ba 100%);
  color: white;

  &:hover {
    background: linear-gradient(90deg, #bc46ba 0%, #ff465d 100%);
    color: white;
  }
`;
