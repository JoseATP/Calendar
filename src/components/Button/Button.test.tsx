import { render, screen, cleanup } from "@testing-library/react";
import Button, { BUTTON_TYPE_CLASSES } from "./Button.component";
import { colors } from "../../constants/colors.constants";

afterEach(cleanup);

describe("Button", () => {
  it("Should render Button component correctly", () => {
    const { asFragment } = render(
      <Button buttonType={BUTTON_TYPE_CLASSES.base} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("check base button style", () => {
    render(<Button buttonType={BUTTON_TYPE_CLASSES.base}>Button</Button>);

    const ButtonElem = screen.getByText("Button");

    expect(ButtonElem).toHaveStyle({ backgroundColor: `${colors.base}` });
  });

  it("check grey button style", () => {
    render(<Button buttonType={BUTTON_TYPE_CLASSES.grayButton}>Button</Button>);

    const ButtonElem = screen.getByText("Button");

    expect(ButtonElem).toHaveStyle({ backgroundColor: `${colors.grey}` });
  });

  it("check red button style", () => {
    render(<Button buttonType={BUTTON_TYPE_CLASSES.redButton}>Button</Button>);

    const ButtonElem = screen.getByText("Button");

    expect(ButtonElem).toHaveStyle({ backgroundColor: `${colors.error}` });
  });

  it("check gradient button style", () => {
    render(
      <Button buttonType={BUTTON_TYPE_CLASSES.gradientButton}>Button</Button>
    );

    const ButtonElem = screen.getByText("Button");

    expect(ButtonElem).toHaveStyle(
      `background: linear-gradient(90deg, ${colors.base2} 0%, ${colors.base} 100%)`
    );
  });

  it("check button disabled", () => {
    render(
      <Button disabled={true} buttonType={BUTTON_TYPE_CLASSES.base}>
        Button
      </Button>
    );

    const ButtonElem = screen.getByText("Button");

    expect(ButtonElem).not.toBeEnabled();
  });

  it("check button enabled", () => {
    render(
      <Button disabled={false} buttonType={BUTTON_TYPE_CLASSES.base}>
        Button
      </Button>
    );

    const ButtonElem = screen.getByText("Button");

    expect(ButtonElem).toBeEnabled();
  });
});
