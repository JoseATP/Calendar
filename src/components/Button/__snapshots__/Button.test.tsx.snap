// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render Button component correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-beySbM haKdpM"
  />
</DocumentFragment>
`;
